src/bp/core/services/auth/auth-service.ts

```diff
@@ -161,22 +192,86 @@ export default class AuthService {
 
   async login(email: string, password: string, newPassword?: string, ipAddress: string = ''): Promise<string> {
     await this.checkUserAuth(email, password, newPassword, ipAddress)
+    const config = await this.configProvider.getBotpressConfig()
     this.stats.track('auth', 'login', 'success')
 
     if (newPassword) {
+      this._validatePassword(newPassword, config)
+
       const hash = saltHashPassword(newPassword)
       await this.updateUser(email, {
         password: hash.hash,
         salt: hash.salt,
-        password_expired: false
+        password_expired: false,
+        ...this._addPasswordExpiry(config)
       })
     }
 
     debug('login', { email, ipAddress })
 
-    await this.updateUser(email, { last_ip: ipAddress }, true)
-    const config = await this.configProvider.getBotpressConfig()
+    await this.updateUser(
+      email,
+      {
+        last_ip: ipAddress,
+        unsuccessful_logins: 0,
+        last_login_attempt: undefined,
+        locked_out: false
+      },
+      true
+    )
+
     const duration = config.jwtToken && config.jwtToken.duration
     return generateUserToken(email, isSuperAdmin(email, config), duration, TOKEN_AUDIENCE)
   }
+
+  private _validatePassword(password: string, config: BotpressConfig) {
+    const authOptions = _.get(config, 'pro.auth.options')
+    if (!authOptions) {
+      return
+    }
+
+    let rules: any = {}
+    if (authOptions.passwordMinLength) {
+      rules.length = { minLength: authOptions.passwordMinLength }
+    }
+
+    if (authOptions.requireComplexPassword) {
```

Comment:
Can a user customize its rules ?

Reply:
Nop, I didn't see really the point, either they enable it or not